src/main/java/pl/agh/edu/iisg/io/vmms/vmmsbackend/controller/ReservationController.java

```diff
@@ -0,0 +1,233 @@
+        return reservationService
+                .getConfirmedOrBeforeDeadlineToConfirmReservationsBetweenDatesForVMPool(vmPoolShortName, from, to)
+                .stream()
+                .map(this::convertToDto)
+                .collect(Collectors.toList());
+    }
+
+    @RequestMapping(path = RESERVATION_ENDPOINT, method = RequestMethod.GET)
+    public ReservationDto getReservation(
+            @RequestParam("reservationId") Long id) throws HttpException {
+        Optional<Reservation> reservation = reservationService.find(id);
+        if (reservation.isPresent()) {
+            return convertToDto(reservation.get());
+        } else {
+            throw new ReservationExpiredException();
+        }
+    }
+
+    @RequestMapping(path = RESERVATION_ENDPOINT, method = RequestMethod.POST)
+    @ResponseStatus(HttpStatus.CREATED)
+    public ReservationResponseDto createReservation(
+            @Valid @RequestBody ReservationRequestDto reservationRequest) {
+        Reservation reservation = convertToReservation(reservationRequest);
+        reservation = reservationService.saveTemporary(
+                reservation,
+                reservationRequest.getStartTime(),
+                reservationRequest.getEndTime(),
+                reservationRequest.getDates());
+
+        List<Date> daysReserved = reservation
+                .getPeriods()
+                .stream()
+                .map(this::convertPeriodToDay)
+                .collect(Collectors.toList());
+
+        List<Date> daysRequested = reservationRequest.getDates();
+        daysRequested.removeAll(daysReserved);
+        return new ReservationResponseDto(reservation.getId(), daysRequested);
+    }
+
+    @RequestMapping(path = CONFIRM_TMP_ENDPOINT, method = RequestMethod.PUT)
+    public String confirmReservation(
+            @RequestBody Long reservationId) throws ReservationExpiredException {
+        Optional<Reservation> reservation = reservationService.findIfNotExpired(reservationId);
+        if (reservation.isPresent()) {
+            reservationService.confirm(reservation.get());
+            return "CONFIRMED";
+        } else {
+            throw new ReservationExpiredException();
+        }
+    }
+
+    @RequestMapping(path = CANCEL_TMP_ENDPOINT, method = RequestMethod.DELETE)
+    public String cancelReservation(
+            @RequestBody Long reservationId) throws ReservationExpiredException {
+        Optional<Reservation> reservation = reservationService.find(reservationId);
+        if (reservation.isPresent()) {
+            reservationService.delete(reservation.get());
+            return "CANCELED";
+        } else {
+            throw new ReservationExpiredException();
+        }
+    }
+
+    @RequestMapping(path = RESERVATION_ENDPOINT, method = RequestMethod.DELETE)
+    public String deleteReservation(
+            @RequestBody Long reservationId) throws ReservationNotFoundException {
```

Comment:
If possible, could we make this a @RequestParam? Even though delete request's body is not explicitly forbidden I think it is more common to use a parameter. Also, our UI library does not support body for DELETE, so It would make things much easier